VMT/KCSA/KCSA-CVE-2020-2025.md

```diff
@@ -0,0 +1,62 @@
+announcement-date: 2020-05-28
+
+id: KCSA-CVE-2020-2025
+
+title: Kata Containers with Cloud Hypervisor guest image persists vulnerability
+
+description: Kata Containers before 1.11.0 on Cloud Hypervisor persists guest
+filesystem changes to the underlying image file on the host. A malicious guest
+can overwrite the image file to gain control of all subsequent guest VMs.
+Since Kata Containers uses the same VM image file with all VMMs, this issue
+may also affect QEMU and Firecracker based guests.
+
+affected-components:
+
+  - components: `kata-runtime`
+    version: Before v1.11.0
+
+vulnerabilities:
+
+  - cve-id: CVE-2020-2025
+
+reporters:
+
+  - name: Yuval Avrahami
+    affiliation: Yuval Avrahami
+    reported:
+      - CVE-2020-2025
+
+issues:
+
+  links:
+    - https://github.com/kata-containers/runtime/issues/2488
+
+ reviews:
+
+  v1.11.0:
+    - https://github.com/kata-containers/runtime/pull/2487
+
+  type: GitHub
+
+reproduce:
+  - Create a Kata container using Cloud Hypervisor
+  - Create a new file on the guest rootfs
+  - View the guest rootfs image on the host and the file can be seen there
+
+notes:
+  - The vulnerability can be used to attack other guests by malicious containers
+    that find other ways to gain control over the guest. And all users running
+    Kata Containers on top of Cloud Hypervisor are recommended to upgrade.
+
+    When running Kata Containers with Cloud Hypervisor, any change made to root
+    filesystem device is written to the underlying .img file. Since the device
+    is plugged as read-write, a malicious guest could write to it through
+    utilities like debugfs.
```

Comment:
I believe this is a mistake from my original advisory - `debugfs` isn't needed if the guest is malicious, the guest can simply create/modify files and the changes will propogate to the image file on the host.

Reply:
@YuvalAvra  `malicious guest` is a container that found a way to escape from the namespaces in the guest?

Reply:
That's how I refer to a guest running malicious code outside the container scope, but perhaps that's not Kata's terminology. It's not necessarily a container that escaped, for example `CVE-2020-2025` would result in subsequent guests being malicious without their containers escaping.

My point here is to separate distinct issues: `CVE-2020-2025` is exploitable from outside the container, if the container has a way to access the guest - that's a different issue.

Reply:
Good point! I've re-phrased it and dropped `debugfs`.